src/app/pages/PerpetualPage/components/TradingChart/datafeed.ts

```diff
@@ -0,0 +1,186 @@
+/**
+ * TradingChart Datafeed
+ *
+ * Implementation of TradingView Charting Library JS API (v18.043):
+ * https://github.com/tradingview/charting_library/wiki/JS-Api/f62fddae9ad1923b9f4c97dbbde1e62ff437b924
+ *
+ * If the version of the library is updated, then modifications may
+ * be necessary to this file and the realtime streaming.ts file in
+ * this directory. Refer to:
+ * https://github.com/tradingview/charting_library/wiki/Breaking-Changes
+ */
+
+import { subscribeOnStream, unsubscribeFromStream } from './streaming';
+import {
+  CandleDuration,
+  CandleDictionary,
+} from '../../hooks/graphql/useGetCandles';
+import {
+  Bar,
+  config,
+  supportedResolutions,
+  resolutionMap,
+  symbolMap,
+  makeApiRequest,
+} from './helpers';
+
+const lastBarsCache = new Map<string, Bar>();
+
+const tradingChartDataFeeds = {
+  // https://github.com/tradingview/charting_library/wiki/JS-Api/f62fddae9ad1923b9f4c97dbbde1e62ff437b924#onreadycallback
+  onReady: callback => {
+    setTimeout(() => callback(config));
+  },
+
+  // https://github.com/tradingview/charting_library/wiki/JS-Api/f62fddae9ad1923b9f4c97dbbde1e62ff437b924#searchsymbolsuserinput-exchange-symboltype-onresultreadycallback
+  searchSymbols: async (
```

Comment:
Do we want to have it here so we know is disabled (I don't think it's necessary)?

Reply:
Removed it and left the comment